docs/walkthrough/using-callbacks.md

```diff
@@ -0,0 +1,53 @@
+(using-callbacks)=
+
+# Using Callbacks
+
+Callbacks are a way of adding additional methods to the finetuning process. The methods are executed when certain events occur and there are several callback classes, each serving a different function by providing different methods for different events.
+A run can be assigned multiple callbacks using the optional `callbacks` parameter when it is created.
+
+## Evaluation
+
+The evaluation callback is used to calculate performance metrics for the model being tuned at the end of each epoch. In order to evaluate the model, two additional data sets - a query dataset and an index dataset - need to be provided as arguments. If no index set is provided, the query dataset is reused instead. Below is an example of the metrics as they are output at the end of finetuning:
```

Comment:
try to explain what is the difference between `eval_data` and `EvaluationCallback`:

`eval_data` is a DocumentArray, used to evaluate machine learning losses, e.g. if you are fine-tuning CLIP, it's mearesuing CLIPLoss.

`EvaluationCallback` is evaluating the "search metrics", such as `average precision` ...

```python
import finetuner

run = finetuner.fit(
    model='some model',
    train_data=MyDocumentArray(..),
    eval_data=MyEvaluatioDocumentArray(...), # eval ML losses
    callbacks=[EvaluationCallback(..)], # evaluating how model is performed on information retrieval metrics.
)
```